src/thread_task.rs

```diff
@@ -991,30 +982,30 @@ pub unsafe extern "C" fn rav1d_worker_task(data: *mut c_void) -> *mut c_void {
 
         'found_unlocked: loop {
             let flush = (*c).flush.load(Ordering::SeqCst);
-            let mut error_0 = (*f).task_thread.error.fetch_or(flush, Ordering::SeqCst) | flush;
+            let mut error_0 = f.task_thread.error.fetch_or(flush, Ordering::SeqCst) | flush;
 
             // run it
             (*tc).f = f;
-            let mut sby = (*t).sby;
-            let mut task_type = (*t).type_0 as c_uint;
+            let mut sby = t.sby;
+            let mut task_type = t.type_0 as c_uint;
             'fallthrough: loop {
                 match task_type {
                     RAV1D_TASK_TYPE_INIT => {
                         if !((*c).n_fc > 1 as c_uint) {
                             unreachable!();
                         }
                         let res = rav1d_decode_frame_init(&mut *f);
```

Comment:
```suggestion
                        let res = rav1d_decode_frame_init(f);
```

I noticed these; these can become `f`s like `(*f)`s became `f`s.

Reply:
Good point, fixed all of them in this function.